.github/workflows/workflow-pr-fork.yml

```diff
@@ -0,0 +1,56 @@
+# Workflow that runs on PR on forks.
+# Triggered by PR events like opening or synchronizing.
+# Build, test, run coverage analysis and release to Maven Central a snapshot release.
+# The job will only run successfully if re-triggered by an internal contributor.
+name: Workflow for PR on forks
+
+on:
+  pull_request_target:
+    types: [ opened, synchronize ]
+
+jobs:
+  access-check:
+    if: ${{ ! startsWith(github.repository, 'TrueLayer/truelayer-java') }}
```

Comment:
This should prevent this WF to run on internal PRs

Reply:
it would be nicer if we could filter on the previous `on` filter, but I could not find a way. The advantage of filtering there is that this WF would not show up as skipped, but not show up at all